tutorials/add_custom_operators/README.md

```diff
@@ -0,0 +1,132 @@
+
+```python
+@OperatorFactory.register
+class _ReductionMeanOperator(_Operator):
+    namespaces = ("MyCustomOpNamespace",)
+    op_type = "ReductionMeanOperator"
+
+    # the value returned by this method will be used as
+    # the constrcutor parameters as is.
+    # In utensor backend, it should return a tuple of string.
+    # Since there is no parameters for `MeanOperator`, an empty tuple is returned
+    @classmethod
+    @must_return_type(Hashable)
+    def get_constructor_parameters(cls, op_info):
+        return tuple()
+
+    # snippet that calls op's constructor and will be placed in the
+    # the initializer list of the model class
+    def get_construct_snippet(self, op_var_name):
+        return OpConstructSnippet(
+            op=self,
+            templ_dtypes=[self.in_dtypes[0]],
+            op_var_name=op_var_name,
+            nested_namespaces=type(self).namespaces,
+        )
+
+    # snippet which declares the op
+    def get_declare_snippet(self, op_var_name, with_const_params=True):
+        return DeclareOpSnippet(
+            op=self,
+            templ_dtypes=[self.in_dtypes[0]],
+            op_var_name=op_var_name,
+            nested_namespaces=type(self).namespaces,
+            with_const_params=with_const_params,
+        )
+
+    # snippet that eval the op
+    def get_eval_snippet(self, op_var_name, op_info, tensor_var_map):
+        return ReductionMeanEvalSnippet(
+            op_info=op_info,
+            templ_dtypes=[self.in_dtypes[0]],
+            op_name=op_var_name,
+            tensor_var_map=tensor_var_map,
+            nested_namespaces=type(self).namespaces,
+        )
+
+```
+
+### Lowering an Operator to our Backend Component
+
+After registering the operator backend component in the OperatorFactory, we need to notify the graph lowering engine to make this target available. Right now, this is a simple registry of operator names and namespaces, but will soon be a proper lowering strategy engine:
+
+```python
+@uTensorRearchGraphLower.CodgenAttributes.register("ReductionMeanOperator")
+def handler(op_info):
+    op_info.code_gen_attributes["namespaces"] = ("MyCustomOpNamespace",)
+```
+
+### Legalizing at the Frontend
+
+It turns out there was a disconnect between the original custom operator naming, `ReductionMeanOperator` and the naming present in the TFLite file `Mean`, so we need to indicate to the Legalizer that these ops need to be renamed:
+
+```python
+# legalize all `Mean` to `ReductionMeanOperator`
+TFLiteLegalizer.register_op_rename(old_name="Mean", new_name="ReductionMeanOperator")
+```
+
+### Registering the new Operator in uTensor CLI
+
+![reduce-model](images/reduceModel.svg)
+
+```bash
+$ utensor-cli --plugin custom_operators convert tf_models/reduceModel.tflite
+```
```

Comment:
```suggestion

Note that `--plugin` is a multiple flag, which means you can load multiple plugins all at once as following:
```

Reply:
```bash
# loading module1 and module2 all at once
$ utensor-cli --plugin module1 --plugin module2 ...
````

Reply:
Updated

Reply:
I don't see the paragraph explaining `--plugin` is multiple.

Reply:
https://github.com/uTensor/utensor_cgen/pull/118/files#diff-f84f8b6742c73039bcf7e6a84dcfa07fR133

Reply:
Ok, then add following snippet as minimal example

```bash
# loading module1 and module2 all at once
$ utensor-cli --plugin module1 --plugin module2 ...
```

Reply:
Done